crypto/crypto/src/hash.rs

```diff
@@ -35,6 +35,25 @@
 //!
 //! # Implementing new hashers
 //!
+//! ## The automatic way
+//!
+//! For any new structure `MyNewStruct` that needs to be hashed, the developer
+//! should use the [`CryptoHasher` derive macro](https://doc.rust-lang.org/reference/procedural-macros.html).
+//!
+//! ```ignore
+//! #[derive(CryptoHasher)]
+//! struct MyNewStruct {
+//!   ...
+//! }
+//! ```
+//!
+//! The macro will define a hasher automatically called `MyNewStructHasher`, and pick a salt
+//! equal to the full module path + "::"  + structure name, i.e. just the
+//! string you would put after `use` in an import statement for `MyNewStruct`. You can then use it
```

Comment:
perhaps we could be a bit more explicit here and state that MyNewStruct in module my-module would result in my_module::MyNewStruct as the salt?

Reply:
I tried! but it seems bors was too fast and sent the commits in w/o my 
920703f